src/server/game/AuctionHouseBot/AuctionHouseBot.cpp

```diff
@@ -272,6 +305,31 @@ char const* AuctionBotConfig::GetHouseTypeName(AuctionHouseType houseType)
     return names[houseType];
 }
 
+// Picks a random character from the list of AHBot chars
+uint32 AuctionBotConfig::GetRandChar() const
+{
+    return _AHBotCharacters[urand(0, _AHBotCharacters.size() - 1)];
+}
+
+// Picks a random AHBot character, but excludes a specific one. This is used
+// to have another character than the auction owner place bids
+uint32 AuctionBotConfig::GetRandCharExclude(uint32 exclude) const
+{
+    // avoid freezing if only one ahbot char (which defeats the purpose but oh well)
+    if (_AHBotCharacters.size() == 1) return GetRandChar();
+
+    uint32 result;
+    do {
+        result = GetRandChar();
+    } while (result == exclude);
```

Comment:
Well, that would work too. As long as the first `{` is moved underneath the `do` command, it's OK.
Both forms have been used in TC source code previously, so I suppose you can use both.